core/src/main/java/com/usebutton/core/data/ConnectionManagerImpl.java

```diff
@@ -89,11 +89,12 @@ public NetworkResponse executeRequest(ApiRequest request) throws ButtonNetworkEx
                 urlConnection.setRequestProperty(entry.getKey(), entry.getValue());
             }
 
-            JSONObject body = request.getBody();
+            JSONObject body = request.getBody() != null ? request.getBody() : new JSONObject();
```

Comment:
There will always be object since we will add app ID etc. The prior assumption in this code was that a body would have already been constructed. This change ensures that we _don't_ make such an assumption.

Reply:
OH somehow I read this as response. 👍